web/api/v1/api_test.go

```diff
@@ -1207,6 +1250,56 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, es storage.E
 				},
 			},
 		},
+		{
+			endpoint: api.query,
+			query: url.Values{
+				"query": []string{
+					`label_replace(vector(42), "foo", "bar", "", "") or label_replace(vector(3.1415), "dings", "bums", "", "")`,
+				},
+				"time": []string{"123.4"},
+			},
+			responseAsJSON: `{
+		"resultType": "vector",
+		"result": [
+			{
+				"metric": {
+					"foo": "bar"
+				},
+				"value": [123.4, "42"]
+			},
+			{
+				"metric": {
+					"dings": "bums"
+				},
+				"value": [123.4, "3.1415"]
+			}
+		]
+	}`,
+		},
+		{
+			endpoint: api.queryRange,
+			query: url.Values{
+				"query": []string{"time()"},
+				"start": []string{"0"},
+				"end":   []string{"2"},
+				"step":  []string{"1"},
+				"limit": []string{"1"},
```

Comment:
Let's repeat this test with `"limit": []string{"2"},` and  `"limit": []string{"0"},`.

Reply:
Since there's no metric returned in this test case, the response for "limit": []string{"2"} and "limit": []string{"0"} will be the same. Is this behavior acceptable? 
Asking, since the `Result` is identical in all three cases.

Reply:
Sorry, I forgot that the `"query"` also has to be changed to the same as above, i.e. it should return two series.

Then limit=0 will return both result because it means "no limits". limit=2 will return both results because it coincides with the limit. and limit=1 will return one result because it got limited.

Reply:
Implemented the changes here https://github.com/prometheus/prometheus/pull/15552/commits/861688da8290315a284e9c8f9165e0e2301bb2c7

PTAL in your own time.
Thanks.